cpp/email_watcher.cc

```diff
@@ -0,0 +1,258 @@
+        unsigned current_line_number(0);
+        while (not input->eof()) {
+            ++current_line_number;
+            const auto line(input->getline());
+            const auto first_equal_pos(line.find('='));
+            if (unlikely(first_equal_pos == std::string::npos or first_equal_pos == 0))
+                LOG_ERROR("missing or leading equal-sign in " + MAPFILE_PATH + " on line #"
+                          + std::to_string(current_line_number) + "!");
+            unsigned last_seen_time;
+            if (unlikely(not StringUtil::ToNumber(line.substr(first_equal_pos + 1), &last_seen_time)))
+                LOG_ERROR("can't convert \"" + line.substr(first_equal_pos + 1) + "\" to an unsigned number! (line #"
+                          + std::to_string(current_line_number) + " in " + MAPFILE_PATH + ")");
+            section_names_to_last_seen_time_map[line.substr(0, first_equal_pos)] = static_cast<time_t>(last_seen_time);
+        }
+    }
+
+    LOG_INFO("Loaded " + std::to_string(section_names_to_last_seen_time_map.size()) + " mappings from " + MAPFILE_PATH + ".");
+
+    return section_names_to_last_seen_time_map;
+}
+
+
+void SaveSectionNamesToLastSeenTimeWindowsMap(const std::map<std::string, time_t> &section_names_to_last_seen_time_map) {
+    const auto output(FileUtil::OpenOutputFileOrDie(MAPFILE_PATH));
+    for (const auto &[section_name, last_seen_time] : section_names_to_last_seen_time_map)
+        (*output) << section_name << '=' << last_seen_time << '\n';
+    LOG_INFO("Wrote " + std::to_string(section_names_to_last_seen_time_map.size()) + " entries/entry to " + MAPFILE_PATH + ".");
+}
+
+
+} // unnamed namespace
+
+
+int Main(int argc, char *argv[]) {
+    if (argc != 3)
+        ::Usage("ini_file_path mbox_path");
+
+    const IniFile ini_file(argv[1]);
+
+    std::vector<std::string> notification_email_addresses;
+    if (StringUtil::SplitThenTrimWhite(ini_file.getString("", "notify"), ',', &notification_email_addresses) == 0)
+        LOG_ERROR("we need at least one notification email address in the \"notify\" entry in \""
+                  + ini_file.getFilename() + "\"!");
+
+    const std::string backup_dir_path(ini_file.getString("", "backup_dir_path") + "/");
+
+    const auto overdue_db(OpenOrCreateOverdueDatabase());
+    const auto email_descriptions(LoadEmailDescriptions(ini_file));
+
+    const std::string mbox_filename(argv[2]);
+    const MBox mbox(mbox_filename);
+    unsigned email_message_count(0);
+    std::vector<std::string> unmatched_emails;
+    std::set<std::string> matched_section_names;
+    for (const auto &email_message : mbox) {
+        ++email_message_count;
+        for (const auto &[section_name, email_description] : email_descriptions) {
+            if (email_description.subjectAndBodyMatched(email_message))
+                matched_section_names.emplace(section_name);
+            else
+                unmatched_emails.emplace_back(email_message.toString());
+        }
+    }
```

Comment:
I think it would be useful to always forward messages hat have either originally been sent with high priority or contain a pre-defined custom header.

Reply:
I think dealing w/ custom headers is difficult and also not necessary.  There are several, possibly many, custom headers in use by email systems.

Reply:
That's why i propose defining our own like e.g. "X-UB-ERROR" which could be added if any program exits with LOG_ERROR and then be automatically redirected whenever this header is found.

Reply:
We can postpone this or just use priority 5.